src/dmd/todt.d

```diff
@@ -456,6 +456,14 @@ extern (C++) void Expression_toDt(Expression e, ref DtBuilder dtb)
         }
     }
 
+    /* https://issues.dlang.org/show_bug.cgi?id=12652
+       Non-constant hash initializers should have a special-case diagnostic
+     */
+    void visitAssocArrayLiteral(AssocArrayLiteralExp e)
+    {
+        e.error("Associative array literals currently cannot be initialized globally, in structs or in classes; try using 'enum' or a module constructor instead");
```

Comment:
Error messages are not capitalized

Reply:
The error message does not quite make sense. The one from the bug report is better:
```
Associative array literals currently cannot be used to initialize globals. Try using a module constructor instead.
```

Reply:
And remove "currently".

Reply:
They aren't actually globals, either. They are statically allocated.